src/notification/notification.service.integration.test.ts

```diff
@@ -1014,4 +1014,59 @@ describe('NotificationService', () => {
             expect(notifications).toHaveLength(0);
         });
     });
+
+    describe('removePostNotifications', () => {
+        it('should remove all notifications for a given post', async () => {
+            const [[aliceAccount, ,], [bobAccount, ,]] = await Promise.all([
+                fixtureManager.createInternalAccount(),
+                fixtureManager.createInternalAccount(),
+            ]);
+
+            const post = await fixtureManager.createPost(aliceAccount);
+
+            // Create multiple notifications for the post
+            await Promise.all([
+                fixtureManager.createNotification(
+                    aliceAccount,
+                    bobAccount,
+                    NotificationType.Like,
+                ),
+                fixtureManager.createNotification(
+                    aliceAccount,
+                    bobAccount,
+                    NotificationType.Repost,
+                ),
+                fixtureManager.createNotification(
+                    aliceAccount,
+                    bobAccount,
+                    NotificationType.Mention,
+                ),
+            ]);
+
+            // Add post_id to the notifications
+            await client('notifications')
```

Comment:
I think `createNotification` should be updated to accept the post the notification is for (if it's a post related notification) - what do you think?

Reply:
Yeah, made things a lot more easier. Added https://github.com/TryGhost/ActivityPub/pull/672/commits/756195618e85042570bef614556b45ac1a552554